homeassistant/components/ezviz/config_flow.py

```diff
@@ -0,0 +1,251 @@
+
+    async def async_step_user(self, user_input=None):
+        """Handle a flow initiated by the user."""
+
+        # Check if ezviz cloud account is present in entry config.
+        # Return camera user flow if present.
+        for item in self._async_current_entries():
+            if item.data.get(CONF_TYPE) == ATTR_TYPE_CLOUD:
+                return await self.async_step_user_camera()
+
+        errors = {}
+
+        if user_input is not None:
+            if CONF_TIMEOUT not in user_input:
+                user_input[CONF_TIMEOUT] = DEFAULT_TIMEOUT
+
+            try:
+                return await self._validate_and_create_auth(user_input)
+
+            except InvalidAuth:
+                errors["base"] = "invalid_auth"
+
+            except AbortFlow:
+                raise
+
+            except Exception:  # pylint: disable=broad-except
+                _LOGGER.exception("Unexpected exception")
+                return self.async_abort(reason="unknown")
+
+        data_schema = vol.Schema(
+            {
+                vol.Required(CONF_USERNAME): str,
+                vol.Required(CONF_PASSWORD): str,
+                vol.Optional(CONF_REGION, default=DEFAULT_REGION): str,
+            }
+        )
+
+        return self.async_show_form(
+            step_id="user", data_schema=data_schema, errors=errors or {}
+        )
+
+    async def async_step_user_camera(self, user_input=None):
+        """Handle a flow initiated by the user."""
+
+        errors = {}
+
+        if user_input is not None:
+            await self.async_set_unique_id(user_input[ATTR_SERIAL])
+            self._abort_if_unique_id_configured()
+
+            return self.async_create_entry(
+                title=user_input[ATTR_SERIAL],
+                data={
+                    CONF_USERNAME: user_input[CONF_USERNAME],
+                    CONF_PASSWORD: user_input[CONF_PASSWORD],
+                    CONF_TYPE: ATTR_TYPE_CAMERA,
+                },
+            )
+
+        camera_schema = vol.Schema(
+            {
+                vol.Required(CONF_USERNAME, default=DEFAULT_CAMERA_USERNAME): str,
+                vol.Required(CONF_PASSWORD): str,
+                vol.Required(ATTR_SERIAL): str,
+            }
+        )
+
+        return self.async_show_form(
+            step_id="user_camera", data_schema=camera_schema, errors=errors or {}
+        )
+
+    async def async_step_discovery(self, discovery_info):
+        """Handle a flow for discovered camera without rtsp config entry."""
+
+        await self.async_set_unique_id(discovery_info[ATTR_SERIAL])
+        self._abort_if_unique_id_configured()
+
+        return await self.async_step_confirm(discovery_info)
+
+    async def async_step_confirm(self, confirm_info):
+        """Confirm and create entry from discovery step."""
+        errors = {}
+
+        if confirm_info is not None:
+            if confirm_info.get(CONF_PASSWORD) is not None:
+                return self.async_create_entry(
```

Comment:
Can we test the credentials before we create the camera entry?

Reply:
It looks like I can test the RTSP creds with ffmpeg.

ffprobe -hide_banner rtsp://admin:<password>@192.168.1.59:554

Would return stream details or "Server returned 401" unauthorized on incorrect password. Do you know of any integrations that I can use as an example?

Reply:
Example of what? I recommend our scaffold script to generate example for credential validation in the config flow.

Reply:
Done. Created module that sends RTSP DESCRIBE messages to camera to verify credentials.